receiver/pluginreceiver/README.md

```diff
@@ -0,0 +1,83 @@
+# Plugin Receiver
+
+The Plugin Receiver is designed to run templated OpenTelemetry pipelines. This allows users to store complex workflows within a plugin that is loaded by the receiver.
+
+Supported pipeline types: `logs`, `metrics`, `traces`
+
+## Configuration
+| Field        | Default | Required | Description |
+| ---          | ---     | ---      | ---         |
+| `path`       |         | `true`   | The path to the plugin file. |
+| `parameters` | { }     | `false`  | A map of `key: value` parameters used to render the plugin's templated pipeline. |
+
+### Example Configuration
+```yaml
+receivers:
+  plugin:
+    path: ./plugins/simplehost.yaml
+    parameters:
+      enable_cpu: false
+      enable_memory: true   
+```
+
+## Plugins
+Plugins are yaml files that define three key aspects:
+- Metadata
+- Parameters
+- Template
+
+### Example
+```yaml
+title: Simple Host Plugin
+description: A plugin utilizing the hostmetrics receiver
+version: 0.0.0
+parameters:
+- name: enable_cpu
+  type: bool
+  default: false
+- name: enable_memory
+  type: bool
+  default: true
+template: |
+  receivers:
+    hostmetrics:
+      scrapers:
+  {{if .enable_cpu}}
+        cpu:
+  {{end}}
+  {{if .enable_memory}}
+        memory:
+  {{end}}
+  service:
+    pipelines:
+      metrics:
+        receivers: [hostmetrics]
+```
+### Metadata
+Metadata fields are used to catalog and distinguish plugins. The following fields are used for metadata:
+- `title`
+- `description`
+- `version`
+
+### Parameters
+Parameters are the fields used to configure a plugin. The values of these fields are used when rendering the plugin's template, resulting in a dynamic pipeline. 
+
+The following keys are used when defining a parameter.
+| Key | Description |
+| --- | --- |
+| `name`      | The name of the parameter. This is the key used when configuring the parameter within the receiver. |
```

Comment:
Are all of these parameters required? If not are there default values?

Reply:
No. They are not required at the moment. The defaults would be the zero values of those fields when unmarshalling into the struct. As stated in the comment above, we could add additional validation if we think it's necessary.

Reply:
I think that's fine maybe just adding a `required` column to the table just so a user knows the minimum set.